src/SFA.DAS.Employer.PR.Web/Controllers/SelectTrainingProviderController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Web.Authentication;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Infrastructure.Services;
+using SFA.DAS.Employer.PR.Web.Models;
+using SFA.DAS.Employer.PR.Web.Models.Session;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers;
+
+[Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
+[Route("accounts/{employerAccountId}/providers/new/selectProvider", Name = RouteNames.SelectTrainingProvider)]
+public class SelectTrainingProviderController(ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitViewModel> _validator) : Controller
+{
+    [HttpGet]
+    public IActionResult Index([FromRoute] string employerAccountId)
+    {
+        var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
+
+        if (sessionModel == null)
+        {
+            return RedirectToAction("Index", "YourTrainingProviders", new { employerAccountId });
+        }
```

Comment:
For now this is fine as we have only one set of data going into app cache and only one usage, but if this changes then we need to refactor this to retrieve it from service that manages cache for this data set.

Reply:
agree